client/CPlayerInterface.cpp

```diff
@@ -211,47 +212,189 @@ void CPlayerInterface::playerStartsTurn(PlayerColor player)
 	}
 }
 
-void CPlayerInterface::performAutosave()
+int extractNumberAfterHash(const std::string& input) 

```

Comment:
1) Either make this function part of PlayerInterface class or mark it as 'static'
2) Perhaps move all autosave-related logic outside of player interface into a separate class? PlayerInterface already has such similar extracted components

Reply:
extractNumberAfterHash has nothing specific to PlayerInterface so I moved it to TextOperations along with the other string operating functions